src/components/pages/doc.js

```diff
@@ -123,7 +123,19 @@ function Doc({ initialState, PageTemplate }) {
     window.removeEventListener('resize', ensureControlPanelPosition);
   });
 
-  const initialView = Object.values(VIEW).find(v => v === request.query.view) || VIEW.display;
+  const determineInitialViewState = () => {
+    const requestView = Object.values(VIEW).find(v => v === request.query.view);
+    if (!requestView) {
+      return { preSetView: null, view: VIEW.display };
+    }
+
+    if (requestView === VIEW.edit) {
```

Comment:
I think this `if` block is not necessary as the controller already prevents unprivileged users from requesting the page in edit mode:

~~~
    if (view === DOC_VIEW_QUERY_PARAM.edit && !canEditDocContent({ user, doc, room })) {
      return res.redirect(routes.getDocUrl({ id: doc._id, slug: doc.slug }));
    }
~~~
src/server/document-controller.js, lines 105-107

Reply:
Nicely spotted, simplified, thanks!